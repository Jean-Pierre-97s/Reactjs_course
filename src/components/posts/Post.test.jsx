/* eslint-disable no-undef */
import { render, screen } from "@testing-library/react";
import { Posts } from ".";

const props = {
  posts: [
    {
      title: "Title 1",
      body: "body 1",
      id: 1,
      cover: "img/img1.png",
    },
    {
      title: "Title 2",
      body: "body 2",
      id: 2,
      cover: "img/img2.png",
    },
    {
      title: "Title 3",
      body: "body 3",
      id: 3,
      cover: "img/img3.png",
    },
  ],
};

describe("<Posts />", () => {
  it("Should render the posts", () => {
    render(<Posts {...props} />);

    expect(screen.getAllByRole("heading", { name: /Title/i })).toHaveLength(3);
    expect(screen.getAllByRole("img", { name: /Title/i })).toHaveLength(3);
    expect(screen.getAllByText(/body/i)).toHaveLength(3);
  });

  it("Should match snapshot", () => {
    const { container } = render(<Posts {...props} />);

    expect(container).toMatchSnapshot();
  });

  it("Should not render posts", () => {
    render(<Posts />);

    expect(screen.queryByRole("heading")).not.toBeInTheDocument();
  });
});
