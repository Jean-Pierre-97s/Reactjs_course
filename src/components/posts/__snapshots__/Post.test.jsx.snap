// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> Should match snapshot 1`] = `
<div>
  <div
    class="posts"
  >
    <div
      class="post"
    >
      <img
        alt="Title 1"
        src="img/img1.png"
      />
      <div
        class="post-content"
      >
        <h2>
          1
           
          Title 1
        </h2>
        <p>
          body 1
        </p>
      </div>
    </div>
    <div
      class="post"
    >
      <img
        alt="Title 2"
        src="img/img2.png"
      />
      <div
        class="post-content"
      >
        <h2>
          2
           
          Title 2
        </h2>
        <p>
          body 2
        </p>
      </div>
    </div>
    <div
      class="post"
    >
      <img
        alt="Title 3"
        src="img/img3.png"
      />
      <div
        class="post-content"
      >
        <h2>
          3
           
          Title 3
        </h2>
        <p>
          body 3
        </p>
      </div>
    </div>
  </div>
</div>
`;
